File: client-libs/typescript/examples/contactImport.example.ts
/** Contact Import DAG Example

Example demonstrating how to create contact import DAGs.
*/

import { createContactImportDAG } from "../src/builders/ContactImportDAGBuilder";
import { ImportExportDAGBuilder } from "../src/builders/UnifiedDAGBuilder";

// Example 1: Basic contact import
export function exampleBasicContactImport() {
  const dag = createContactImportDAG("my-uploads-bucket", "uploads/contacts_2024.csv");

  console.log("Contact Import DAG:", JSON.stringify(dag, null, 2));
  return dag;
}

// Example 2: Contact import with custom retry settings
export function exampleCustomRetry() {
  const dag = createContactImportDAG("my-uploads-bucket", "uploads/large_contacts_2024.csv", {
    retryCount: 5,
    retryInterval: 10, // 10 minutes between retries
    customVars: {
      source: "manual_upload",
      user_id: "user-123",
    },
  });

  console.log("Contact Import DAG (Custom Retry):", JSON.stringify(dag, null, 2));
  return dag;
}

// Example 3: Using unified builder
export function exampleUnifiedBuilder() {
  const dag = ImportExportDAGBuilder.createImportDAG(
    "contact",
    "my-uploads-bucket",
    "uploads/contacts_2024.csv",
    {
      workflowId: "my-custom-workflow-id",
    }
  );

  console.log("Contact Import DAG (Unified Builder):", JSON.stringify(dag, null, 2));
  return dag;
}

if (require.main === module) {
  console.log("=".repeat(60));
  console.log("Example 1: Basic Contact Import");
  console.log("=".repeat(60));
  exampleBasicContactImport();

  console.log("\n" + "=".repeat(60));
  console.log("Example 2: Custom Retry Settings");
  console.log("=".repeat(60));
  exampleCustomRetry();

  console.log("\n" + "=".repeat(60));
  console.log("Example 3: Unified Builder");
  console.log("=".repeat(60));
  exampleUnifiedBuilder();

  console.log("\n" + "=".repeat(60));
  console.log("Examples complete!");
  console.log("=".repeat(60));
}

File: client-libs/python/examples/contact_import_example.py
"""Contact Import DAG Example

Example demonstrating how to create contact import DAGs.
"""

import sys
from pathlib import Path

# Add parent directory to path for imports
sys.path.insert(0, str(Path(__file__).parent.parent))

from dag_builders.contact_import import create_contact_import_dag

# Example 1: Basic contact import
def example_basic_contact_import():
    """Create a basic contact import DAG"""
    dag = create_contact_import_dag(
        s3_bucket="my-uploads-bucket",
        s3_key="uploads/contacts_2024.csv",
    )

    print("Contact Import DAG:")
    print(dag)
    return dag


# Example 2: Contact import with custom retry settings
def example_custom_retry():
    """Create contact import DAG with custom retry configuration"""
    dag = create_contact_import_dag(
        s3_bucket="my-uploads-bucket",
        s3_key="uploads/large_contacts_2024.csv",
        retry_count=5,
        retry_interval=10,  # 10 minutes between retries
        custom_vars={
            "source": "manual_upload",
            "user_id": "user-123",
        },
    )

    print("Contact Import DAG (Custom Retry):")
    print(dag)
    return dag


# Example 3: Using unified builder
def example_unified_builder():
    """Create contact import using unified builder"""
    from dag_builders.unified import ImportExportDAGBuilder

    dag = ImportExportDAGBuilder.create_import_dag(
        service="contact",
        s3_bucket="my-uploads-bucket",
        s3_key="uploads/contacts_2024.csv",
        workflow_id="my-custom-workflow-id",
    )

    print("Contact Import DAG (Unified Builder):")
    print(dag)
    return dag


if __name__ == "__main__":
    print("=" * 60)
    print("Example 1: Basic Contact Import")
    print("=" * 60)
    example_basic_contact_import()

    print("\n" + "=" * 60)
    print("Example 2: Custom Retry Settings")
    print("=" * 60)
    example_custom_retry()

    print("\n" + "=" * 60)
    print("Example 3: Unified Builder")
    print("=" * 60)
    example_unified_builder()

    print("\n" + "=" * 60)
    print("Examples complete!")
    print("=" * 60)
